### PR TITLE
changes in validation message

### PR DESCRIPTION
**Pull request by ShrutiSachpara**

### CODE REVIEW

**Review by krishna2919 — APPROVED** [review]

File: src/app/forgot-password/forgot-password.component.html [hunk 1]
```diff
@@ -19,18 +19,30 @@ <h4 class="mt-5 font-weight-bold">Forgot Password?</h4>
...
                 [isEmail]="true"
-                [errorMessage]="emailIncorrect"
                 class="form-group mb-3"
               ></app-base-input>
 
+              <div
+                *ngIf="emailControl.hasError('pattern') && emailControl.touched"
+                class="text-danger"
+              >
+                Please enter a valid email address.
```
Comment (SwetaaRampariya) [inline src/app/forgot-password/forgot-password.component.html]: this message should be come from constants file.

in constants file you can add all the strings which you need in future as well.

File: src/app/forgot-password/forgot-password.component.html [hunk 2]
```diff
@@ -19,18 +19,30 @@ <h4 class="mt-5 font-weight-bold">Forgot Password?</h4>
...
                 [control]="otpControl"
-                [errorMessage]="validOtp"
                 class="form-group mb-3"
               ></app-base-input>
 
+              <div
+                *ngIf="otpControl.hasError('pattern') && otpControl.touched"
+                class="text-danger"
+              >
+                Please enter a valid otp.
```
Comment (SwetaaRampariya) [inline src/app/forgot-password/forgot-password.component.html]: this message should be come from constants file.

File: src/app/forgot-password/forgot-password.component.html [hunk 3]
```diff
@@ -40,6 +52,16 @@ <h4 class="mt-5 font-weight-bold">Forgot Password?</h4>
...
               ></app-base-input>
 
+              <div
+                *ngIf="
+                  newPasswordControl.hasError('pattern') &&
+                  newPasswordControl.touched
+                "
+                class="text-danger"
+              >
+                Please enter a valid password.
```
Comment (SwetaaRampariya) [inline src/app/forgot-password/forgot-password.component.html]: this message should be come from constants file.

File: src/app/forgot-password/forgot-password.component.html [hunk 4]
```diff
@@ -48,21 +70,31 @@ <h4 class="mt-5 font-weight-bold">Forgot Password?</h4>
...
               ></app-base-input>
+
+              <div
+                *ngIf="
+                  confirmPasswordControl.hasError('pattern') &&
+                  confirmPasswordControl.touched
+                "
+                class="text-danger"
+              >
+                Please enter a valid password.
```
Comment (SwetaaRampariya) [inline src/app/forgot-password/forgot-password.component.html]: this message should be come from constants file.

File: src/app/forgot-password/forgot-password.component.html [hunk 5]
```diff
@@ -48,21 +70,31 @@ <h4 class="mt-5 font-weight-bold">Forgot Password?</h4>
...
                 {{ showOtpInput ? "Verify OTP" : "Send" }}
               </app-base-button>
               <a
                 *ngIf="!showOtpInput"
-                routerLink="/login"
                 class="btn btn-light btn-lg"
+                [routerLink]="'/login'"
               >
-                <i class="bx bx-arrow-back me-1"></i>Back to Login
+                <span class="bx bx-arrow-back me-1"></span>Back to Login
```
Comment (SwetaaRampariya) [inline src/app/forgot-password/forgot-password.component.html]: this message should be come from constants file.

File: src/app/forgot-password/forgot-password.component.ts [hunk 6]
```diff
@@ -35,7 +31,26 @@ export class ForgotPasswordComponent {
   ) {
     this.emailControl = new FormControl('', [
       Validators.required,
-      Validators.email,
+      Validators.pattern('^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,4}$'),
```
Comment (SwetaaRampariya) [inline src/app/forgot-password/forgot-password.component.ts]: do not add regex here .

File: src/app/login/login.component.ts [hunk 7]
```diff
@@ -30,39 +30,51 @@ export class LoginComponent {
...
-          this.authService.setLoggedIn(true);
-        } else {
-          this.showMessage(response.message, 'dismiss');
-        }
-      });
+      this.loading = true;
+
+      this.loginService
+        .login(data)
+        .subscribe((response: any) => {
```
Comment (SwetaaRampariya) [inline src/app/login/login.component.ts]: do not use any type.

**Review by SwetaaRampariya — CHANGES_REQUESTED** [review]

**Pull request merged**